script.js

```diff
@@ -16,31 +17,40 @@ const getHP = async (browser, player) => {
   await page.goto(player.url)
 
   // wait for the current health element to be visible
-  await page.waitForSelector('.ct-status-summary-mobile__hp-current', { visible: true })
+  // note we're selecting for the mobile view, which is what shows up at the puppeteer default
+  // viewport; we could resize this if required.
+  await page.waitForSelector('.ct-status-summary-mobile__hp-current', {
+    visible: true,
+  })
 
-  const current = await page.$eval('.ct-status-summary-mobile__hp-current', el => el.innerText)
-  const max = await page.$eval('.ct-status-summary-mobile__hp-max', el => el.innerText)
+  const current = await page.$eval(
+    '.ct-status-summary-mobile__hp-current',
+    (el) => el.innerText
+  )
+  const max = await page.$eval(
+    '.ct-status-summary-mobile__hp-max',
+    (el) => el.innerText
+  )
   const half = Math.floor(parseInt(max) / 2)
 
-  return {...player, current, max, half}
+  return { ...player, current, max, half }
 }
 
-(async () => {
+;(async () => {
```

Comment:
What's the semi-colon doing in the syntax here?

Reply:
This gets added by prettier to stop the IIFE being executed as an argument to any preceding function:

```js
mybrokenfunction = function(){

} //no semicolon here
(function(g){


})(this);
```

actually becomes

```js
mybrokenfunction = function(){}(function(g){})(this);
```

Automatic Semicolon Insertion (ASI) is a Whole Thing

(source: https://stackoverflow.com/a/17978926)

Reply:
Ah!

And we need to make a function so we can mark it async AIUI?  The alternative would be `const foo = async () …` and the invoking it?

Reply:
exactly

Reply:
also actually true even if it wasn't async, an IIFE is just a way of immediately running a thing without a separate invocation:

```js
(function() {
    console.log('foo')
})()

// vs

function logFoo() {
    console.log('foo')
}
logFoo()
```